Makefile

```diff
@@ -602,12 +602,20 @@ docker-build-all: $(addprefix docker-build-,$(ALL_ARCH)) ## Build docker images
 docker-build-%:
 	$(MAKE) ARCH=$* docker-build
 
-ALL_DOCKER_BUILD = core kubeadm-bootstrap kubeadm-control-plane clusterctl
+ALL_DOCKER_BUILD = core kubeadm-bootstrap kubeadm-control-plane capd test-extension clusterctl
 
 .PHONY: docker-build
-docker-build: docker-pull-prerequisites ## Run docker-build-* targets for all providers
+docker-build: docker-pull-prerequisites ## Run docker-build-* targets for all the images
 	$(MAKE) ARCH=$(ARCH) $(addprefix docker-build-,$(ALL_DOCKER_BUILD))
 
+ALL_DOCKER_BUILD_E2E = core kubeadm-bootstrap kubeadm-control-plane capd test-extension
+
+.PHONY: docker-build-e2e ## Run docker-build-* targets for all the images required by E2E tests
+docker-build-e2e: ## Run docker-build-* targets for all the images with settings to be used for the e2e tests
+    # please ensure the generated image name matches image names used in the E2E_CONF_FILE;
+    # also the same settings must exist in ci-e2e-lib.sh, capi:buildDockerImage func.
+	$(MAKE) REGISTRY=gcr.io/k8s-staging-cluster-api PULL_POLICY=IfNotPresent TAG=dev $(addprefix docker-build-,$(ALL_DOCKER_BUILD_E2E))
```

Comment:
Should maybe some of these be variables. For example registry might be useful to be able to set?

Reply:
Might be good to use the REGISTRY, PULL_POLICY and TAG vars (REGISTRY and TAG are still set in ci-e2e-lib.sh, PULL_POLICY would have to be re-introduced)

I would propose to set all 3 in ci-e2e-lib.sh and use variables here

Reply:
> Should maybe some of these be variables. For example registry might be useful to be able to set?

This is not necessary because E2E doesn't really rely on a registry but it relies on local images, we just need to agree on a name for the images (+ (IfNotPresent to disable registry lookup),

> Might be good to use the REGISTRY, PULL_POLICY and TAG vars (REGISTRY and TAG are still set in ci-e2e-lib.sh, PULL_POLICY would have to be re-introduced)
>
> I would propose to set all 3 in ci-e2e-lib.sh and use variables here

We can't set vars in ci-e2e-lib.sh only because then it won't work when calling make docker-build-e2e directly

Rephrasing a little bit, we are required to keep settings consistent in 3 places, but according to how we test, there could be different entry points (script-->make-->config, make-->config and also tilt/IDE-->config).
Assuming that it is not worth to over-engineer this, I think that having values hard coded + one comment is fine for now

Reply:
Sounds good. Also not having to export the variables sounds good to me.(usually I had to do that when building images locally to run e2e tests locally, with this target now I don't have to anymore)